### PR TITLE
14 improve template editor

### PR DESCRIPTION
**Pull request by mvgeorgescu**
I made some changes in this PR to update the template editor. The previous implementation was using a package called `markdown-to-jsx` which was able to render our library of MDX components, but it could not render jsx variables (e.g. something like `{props.doc.state_select?.value}`). Instead I switched to using the `mdx-js` and jsx-runtime which is what is also used on our templated pages, so the same thing should be going on under the hood now when running prebuilt mdx templates and templates made in the editor.

I'm leaving this in draft because there is one issue that I need some assistance on. Currently mdx-js will recompile and update mdx on the template editor page when there is ANY change to the textarea field. I would like to change this so that compilation and update happens when the submit button is pressed. I've fiddled with this for a while, but I'm not familiar enough with the react component lifecycle to get the component to update the way that I'd like it to. Any help on this aspect would be greatly appreciated @jackcruz53 @RaeChen07

### CODE REVIEW

File: src/components/editor_flexbox.tsx
Comment (charliepnnl) [inline src/components/editor_flexbox.tsx:54]: I suggest calling this something more description, such as `initialTextAreaContent`.

File: src/components/editor_flexbox.tsx
Comment (charliepnnl) [inline src/components/editor_flexbox.tsx:87]: I think you need to add another state variable for `TemplateView`. 
```
const [TemplateView, setTemplateView] = useState(generateTemplateView(initialTemplateText))
```

File: src/components/editor_flexbox.tsx
Comment (charliepnnl) [inline src/components/editor_flexbox.tsx:77]: `templateText` might be a little more descriptive for our context.

File: src/components/editor_flexbox.tsx
Comment (charliepnnl) [inline src/components/editor_flexbox.tsx:79]: I suggest abstracting this into a function (don't forget to provide jsDoc documentation for it):
```
function generateTemplateView(templateText) {
...
return templateView
}
```
Then this function can be called to generate the initial view AND the updated view.

File: src/components/editor_flexbox.tsx
Comment (charliepnnl) [inline src/components/editor_flexbox.tsx:88]: Given my other suggestions, this method would become:
```
const handleChange = () => {
    templateView = generateTemplateView(templateText)
    setTemplateView = templateView
}
```

File: src/components/editor_flexbox.tsx
Comment (charliepnnl) [inline src/components/editor_flexbox.tsx:123]: Given my suggestions above, `Content` would be replaced with `TemplateView`.

File: src/__tests__/test.test.tsx
Comment (charliepnnl) [inline src/__tests__/test.test.tsx:8]: Nice!

File: src/components/editor.tsx
Comment (charliepnnl) [inline src/components/editor.tsx:10]: Should this h1 be "Template Editor"?

File: src/components/editor_flexbox.tsx
Comment (charliepnnl) [inline src/components/editor_flexbox.tsx:99]: Do we really need to tell it 'playground' is a string? Maybe there is an issue with the implementation of `<StoreProvider>`.

File: src/components/home.tsx
Comment (charliepnnl) [inline src/components/home.tsx:27]: For production, we won't want the editor listed on the home page.  We don't want the average user to know about the editor. The user will have to explicitly change the url to access the editor (an Easter egg).

**Review by charliepnnl — APPROVED** [review]
This is great!  Just some minor comments here to address.

We also discussed some further enhancements in our last standup. I will leave it up to you whether to merge this in and put the enhancements in a separate branch, or put the enhancements here. Either way, the enhancements are not a high priority at this time.

File: src/components/editor_flexbox.tsx
Comment (mvgeorgescu) [reply src/components/editor_flexbox.tsx]: You're right. I think originally I added this because it was producing a type error, but that doesn't seem to be the case anymore in this version.

**Comment by mvgeorgescu** [comment]
Hi @charliepnnl - re-requesting your review on this one. I've gone through a resolved comments you left. I also update the template viewer so that an error prompt display if there is a syntax error when re-rendering a template

File: src/components/editor_flexbox.tsx
Comment (charliepnnl) [inline src/components/editor_flexbox.tsx:98]: How about creating a function `handleTemplateTextChange()` that is called whenever the template text changes. This method calls `setTemplateText()` and executes `localStorage.setItem("templateText", templateText)`. That way the localStorage is updated on every keystroke.

Then there is no need to update localStorage here.

With the ErrorBondary suggestion I make below, the try/catch here can be eliminated.

File: src/components/editor_flexbox.tsx
Comment (charliepnnl) [inline src/components/editor_flexbox.tsx:94]: Is it possible to use a state variable  for `MDXContent` instead? It seems that would simplify logic elsewhere a bit.

File: src/components/editor_flexbox.tsx
Comment (charliepnnl) [inline src/components/editor_flexbox.tsx:105]: if we store MDXContent in state as I suggest above, I don't think we'll need the `useEffect()` method.

**Review by charliepnnl — CHANGES_REQUESTED** [review]
I believe the issue we have with the app crashing and showing an error can be fixed with a well-placed ErrorBoundary.

File: src/components/editor_flexbox.tsx
Comment (mvgeorgescu) [reply src/components/editor_flexbox.tsx:94]: When I try and set the state variable to `mdxComponent.default` (i.e. MDXContent), I get the following:

```
Argument of type 'MDXContent' is not assignable to parameter of type 'Element | (() => Element)'.
  Type 'MDXContent' is not assignable to type '() => Element'.
    Target signature provides too few arguments. Expected 1 or more, but got 0.ts(2345)
```

This error occurs even if I update `generateTemplateView` to be defined in a way that a valid MDXComponent will always be generated.

File: src/components/editor_flexbox.tsx
Comment (mvgeorgescu) [reply src/components/editor_flexbox.tsx]: After doing some experimenting, it doesn't look like the useEffect was needed at all

**Comment by mvgeorgescu** [comment]
Hi @charliepnnl - I'd like to see if we can hop on a call to address the unresolved comments. I made some updates so that the MDXComponent will re render as a user types (assuming valid MDX is provided). This was done by adding an additional handler (`handleSetText`), so now there are vents for `handleSetText` and `handleButtonClick`. We could probably do away with one, but I want to get your thoughts on it.

File: src/components/display_error_error_boundary.tsx
Comment (charliepnnl) [inline src/components/display_error_error_boundary.tsx:31]: Can we also include `error.message` here?

File: src/components/editor_flexbox.tsx
Comment (charliepnnl) [inline src/components/editor_flexbox.tsx:154]: Please either remove this commented out text or provide an explanation (in the comment) for why it is being kept.

File: src/components/radio_wrapper.tsx
Comment (charliepnnl) [inline src/components/radio_wrapper.tsx:29]: For compatibility with Sudha's PR, `doc` should be replace with `data`, here and when setting `value`.

File: src/templates/qa_hpwh.mdx
Comment (charliepnnl) [inline src/templates/qa_hpwh.mdx:1]: For compatibility with Sudha's PR, all `doc`s should be replaced with `data`s.

File: src/templates/qa_hpwh.mdx
Comment (charliepnnl) [inline src/templates/qa_hpwh.mdx:275]: I'm not sure what purpose the `<br/>`s are serving, but we should not be trying to force the layout in this way from the template. We _could_ create a new component `<PageBreak>` that has its styling set to page break before it.

You can leave it this way for now, if needed. I have created a new issue for adding a `<PageBreak>` component.

File: src/templates/qa_hpwh.mdx
Comment (charliepnnl) [inline src/templates/qa_hpwh.mdx:342]: This text is inappropriate for the report since it refers to using the photo input.

File: src/templates/qa_hpwh.mdx
Comment (charliepnnl) [inline src/templates/qa_hpwh.mdx:428]: Oh oh, do we have these images?  I thought we had them before.

**Review by charliepnnl — DISMISSED** [review]
Just a few comments to consider before merging.

File: src/templates/qa_hpwh.mdx
Comment (mvgeorgescu) [reply src/templates/qa_hpwh.mdx:428]: I haven't noticed this before. I've made no changes to the HPWH Selection – Instructions collapsable

**Pull request merged**
